include/MeshAmalgamationBase.h

```diff
@@ -0,0 +1,46 @@
+#include <cstdlib>
+#include <vector>
+#include <stdexcept>
+
+#include "libmesh/dof_map.h"
+#include "libmesh/elem.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/libmesh.h"
+#include "libmesh/linear_implicit_system.h"
+#include "libmesh/mesh.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/system.h"
```

Comment:
Since you don't actually invoke the interface for any of the classes included here, you probably don't need to include their headers.  Instead you can [forward declare](https://www.geeksforgeeks.org/what-are-forward-declarations-in-c/) the classes if necessary.

Reply:
If you do get rid of them here, you'll need to include them in the CPP file instead, where you do invoke the interface.